frontend/components/NavigationBar.js

```diff
@@ -0,0 +1,380 @@
+import Image from "next/image";
+import styles from "../styles/Navbar.module.css";
+import CompactNavigationBar from "./CompactNavigationBar";
+
+// import dropdownContainer icons
+import ChromeReaderModeOutlinedIcon from "@material-ui/icons/ChromeReaderModeOutlined";
+import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
+import PeopleOutlineIcon from "@material-ui/icons/PeopleOutline";
+import PhoneOutlinedIcon from "@material-ui/icons/PhoneOutlined";
+import InsertDriveFileOutlinedIcon from "@material-ui/icons/InsertDriveFileOutlined";
+import HeadsetMicOutlinedIcon from "@material-ui/icons/HeadsetMicOutlined";
+import LocalPrintshopOutlinedIcon from "@material-ui/icons/LocalPrintshopOutlined";
+import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
+import VideocamOutlinedIcon from "@material-ui/icons/VideocamOutlined";
+
+const NavigationBar = () => {
+  const [clearNavBar, setClearNavBar] = useState(false);
+  const [hiddenNavBar, setHiddenNavBar] = useState(false);
+  const [compactNavBar, setCompactNavBar] = useState(false);
+  const [darkMode, setDarkMode] = useState(false);
+  const router = useRouter();
+
+  /* check whether device is in light or dark mode 
+     and change darkMode state accordingly */
+  useEffect(() => {
+    if (
+      window.matchMedia &&
+      window.matchMedia("(prefers-color-scheme: dark)").matches
+    ) {
+      setDarkMode(true);
+    }
+  }, []);
+
+  /* check whether user is at the bottom of the page when scrolling
+     and change hiddenNavBar state accordingly */
+  useEffect(() => {
+    window.addEventListener("scroll", checkBottomScreen);
+    return () => {
+      window.removeEventListener("scroll", checkBottomScreen);
+    };
+  }, []);
+
+  /* check window width when the window is resized
+     and change compactNavBar state accordingly */
+  useEffect(() => {
+    checkWindowWidth();
+    window.addEventListener("resize", checkWindowWidth);
+    return () => {
+      window.removeEventListener("resize", checkWindowWidth);
+    };
+  }, []);
+
+  /* check whether user is at the top of the page when scrolling
+     and change clearNavBar state accordingly */
+  useEffect(() => {
+    checkTopScreen();
+    window.addEventListener("scroll", checkTopScreen);
+    return () => {
+      window.removeEventListener("scroll", checkTopScreen);
+    };
+  }, []);
+
+  /* add mouseover and mouseleave event listeners to the dropdown containers */
+  useEffect(() => {
+    let aboutUsDropdown = document.getElementById("aboutUsDropdown");
+    let mediaDropdown = document.getElementById("mediaDropdown");
+    aboutUsDropdown.addEventListener("mouseover", changeAboutUsToArrowDown);
+    aboutUsDropdown.addEventListener("mouseleave", changeAboutUsToArrowRight);
+    mediaDropdown.addEventListener("mouseover", changeMediaToArrowDown);
+    mediaDropdown.addEventListener("mouseleave", changeMediaToArrowRight);
+  }, []);
+
+  const checkWindowWidth = () => {
```

Comment:
Consider using `{ isMobile }` from `react-device-detect` instead of manually checking the window width